src/SpineGenerator.ts

```diff
@@ -348,15 +323,20 @@ export class SpineGenerator {
       const key = `${subTexture.instanceId}_${blendMode}`;
       let material = SpineAnimationRenderer._materialCache.get(key);
       if (!material) {
-        material = this._createMaterialForTexture(subTexture, engine, blendMode, premultipliedAlpha);
+        material = this._createMaterialForTexture(
```

Comment:
```typescript
if (curLen < lastLen) {
    for (let i = curLen; i < lastLen; i++) {
      const item = _subPrimitives[i];
      subPrimitivePool.return(item);
    }
  }
```

The if condition in the above code is not needed

```typescript
let subTexture = texture.texture;
if (_separateSlotTextureMap.has(slotName)) {
  subTexture = _separateSlotTextureMap.get(slotName);
}
```

should opt

```typescript
let subTexture = _separateSlotTextureMap.get(slotName) || texture.texture;
```

Reply:
done